python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -143,6 +138,12 @@ def __new__(
                 day_offset=day_offset or 0,
             )
 
+        if not is_valid_cron_schedule(cron_schedule):
```

Comment:
before, if an invalid value was provided to partitioned config (i.e. day offset = 8  in a weekly partitioned config), Dagster would throw an error when Dagit loads the partition page or the daemon loads a schedule that contains the partitioned config. 

This bumps this error to instead be thrown at definition time.